### ISSUE TITLE
[bug] cmake_find_package_multi generator uses wrong libs of components

### ISSUE THREAD

**Issue by steffenroeber** [conan-io/conan#7489]
### Environment Details (include every applicable attribute)
  * Win 10
  * Visual Studio 2015
  * Conan version: 1.28.0
  * Python version: 3.8.2

### Steps to reproduce (Include if Applicable)
```python
class CassandraConan(ConanFile):
    name = "cassandra"
...
  def package_info(self):
    self.cpp_info.components["cbl"].libs = ["cbl"]
    self.cpp_info.components["cbl"].libdirs = ["lib%s" % self.settings.build_type]
```

Consumer CMAKELIST:

```cmake
...
target_link_libraries(Test PRIVATE cassandra::cbl)
...
```

We have our libs in different folders "Debug"  and "Release" with identical names. If I use components together with cmake_find_package_multi , then generated cmake/vcxproj files that consume this recipe use same libs for debug and release.
I get something like: Skipping already existing target: CONAN_LIB::cassandra_cbl_cbl. It seems their is something wrong with names of different build types.

**Comment by memsharded** [comment]
Hi @steffenroeber 

I have tried to reproduce, please find the source code here:
[test_project.zip](https://github.com/conan-io/conan/files/5021875/test_project.zip)

```bash
# FIrst create hello, both release and debug
$ conan create .
$ conan create . -s build_type=Debug

$ cd consumer
$ mkdir build && cd build
$ conan install ..
$ conan install .. -s build_type=Debug
$ cmake .. -G "Visual Studio 15 Win64"
-- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.18363.
-- The CXX compiler identification is MSVC 19.16.27032.1
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Library hello found C:/Users/memsharded/.conan/data/hello/0.1/_/_/package/8cf01e2f50fcd6b63525e70584df0326550364e1/libDebug/hello.lib
-- Found: C:/Users/memsharded/.conan/data/hello/0.1/_/_/package/8cf01e2f50fcd6b63525e70584df0326550364e1/libDebug/hello.lib
-- Library hello found C:/Users/memsharded/.conan/data/hello/0.1/_/_/package/8cf01e2f50fcd6b63525e70584df0326550364e1/libDebug/hello.lib
-- Found: C:/Users/memsharded/.conan/data/hello/0.1/_/_/package/8cf01e2f50fcd6b63525e70584df0326550364e1/libDebug/hello.lib
-- Library hello found C:/Users/memsharded/.conan/data/hello/0.1/_/_/package/6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7/libRelease/hello.lib
-- Found: C:/Users/memsharded/.conan/data/hello/0.1/_/_/package/6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7/libRelease/hello.lib
-- Library hello found C:/Users/memsharded/.conan/data/hello/0.1/_/_/package/6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7/libRelease/hello.lib
-- Skipping already existing target: CONAN_LIB::hello_hellolib_hello
-- Found: C:/Users/memsharded/.conan/data/hello/0.1/_/_/package/6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7/libRelease/hello.lib
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/memsharded/conanws/kk/test_package/build
```


Seems to be working ok. Could you please have a look? Thanks!

**Comment by steffenroeber** [comment]
Hi, thanks for fast response. Your example shows exactly my problem. In the generated example.vcxproj all (Debug and Release) sections <Link><AdditionalDependencies> are idendical. 
And you can also see the warning: Skipping already existing target

**Comment by memsharded** [comment]
Thanks for the clarification, as it was correctly finding both, I thought it was ok, I missed the "skipping" part. I'll have another look.

**Comment by memsharded** [comment]
I was able to reproduce, indeed a bug, due to some parallel changes to the cmake code. Proposing a PR https://github.com/conan-io/conan/pull/7492 to fix it for next patch 1.28.1

**Comment by jgsogo** [comment]
Closed via https://github.com/conan-io/conan/pull/7492
